### PR TITLE
update proto for snapshot

### LINKED ISSUE #270: Update snapshot protos to allow new importing and exporting of the tree
# Task Overview

To fix snapshots via statesync in core, the structure of the snapshot protos need to change given the new form of a `NodeHash`

### PR DESCRIPTION
**Pull request by wwestgarth**
closes #270 

see https://github.com/vegaprotocol/vega/pull/4702 for more details

### CODE REVIEW

File: sources/vega/snapshot/v1/snapshot.proto [hunk 1]
```diff
@@ -31,11 +31,14 @@ message Snapshot {
...
 }
 
+// NodeHash represents an exported node from the AVL tree and contains all the data necessary to recontruct and identical copy of the 
+// AVL tree after we've serialised it
 message NodeHash {
-  string full_key = 1; // namespace<separator>key
-  string namespace = 2;
   string key = 3;
   string hash = 4;
+  int32 height = 5;
```
Comment (ze97286) [inline sources/vega/snapshot/v1/snapshot.proto:39]: is height not int64?

Comment (wwestgarth) [reply sources/vega/snapshot/v1/snapshot.proto:39]: This isn't the block height, its the height of the node in the AVL tree. All of the fields here are related to the nodes in the AVL tree. There are lots of comments explaining all these fields on the core side.

Comment (ze97286) [reply sources/vega/snapshot/v1/snapshot.proto:39]: Thanks. might be worth having them here too for auto generation of docs.

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,8 @@
 - [250](https://github.com/vegaprotocol/protos/pull/250) - Rename transaction hash to tx hash 
 - [253](https://github.com/vegaprotocol/protos/pull/253) - Allow `OracleSpec` for internal oracle to be defined without public keys
 - [261](https://github.com/vegaprotocol/protos/pull/261) - Add `staking_asset_total_supply` to staking accounts snapshot
+- [263](https://github.com/vegaprotocol/protos/pull/263) - Add new validator commands
```
Comment (ze97286) [inline CHANGELOG.md:32]: this shouldn't be here - we reverted it...

File: sources/vega/snapshot/v1/snapshot.proto [hunk 3]
```diff
@@ -31,11 +31,14 @@ message Snapshot {
...
+  int64 version = 6;
+  bool is_leaf = 8;
```
Comment (ze97286) [inline sources/vega/snapshot/v1/snapshot.proto:41]: also, silly question, but as this is new, why skip 7?

Comment (wwestgarth) [reply sources/vega/snapshot/v1/snapshot.proto]: Ah because I added some fields, then later removed some and forgot to update. 7 now exists.

**Review by ze97286 — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**